forc/src/cli/commands/completions.rs

```diff
@@ -11,16 +46,138 @@ pub struct Command {
-    let mut cmd = super::super::Opt::command();
-    print_completions(command.shell, &mut cmd);
+    let target = command.target.unwrap_or_else(|| {
+        if let Ok(shell) = std::env::var("SHELL") {
+            match Path::new(shell.as_str())
+                .file_name()
+                .unwrap()
+                .to_str()
+                .unwrap()
+            {
+                "bash" => Target::Bash,
+                "zsh" => Target::Zsh,
+                "fish" => Target::Fish,
+                "pwsh" => Target::PowerShell,
+                "elvish" => Target::Elvish,
+                _ => Target::Bash,
+            }
+        } else {
+            Target::Bash
+        }
+    });
+
+    // Where to store the autocomplete script, ZSH requires a special path, the other shells are
+    // stored in the same path and they are referenced form their respective config files
+    let dir = home::home_dir().map(|p| p.display().to_string()).unwrap();
+    let forc_autocomplete_path = match target {
+        Target::Zsh => {
+            let x = std::process::Command::new("zsh")
+                .arg("-c")
+                .arg("echo $fpath")
+                .output()
+                .expect("Cannot read $FPATH env variable")
+                .stdout;
+            let paths = String::from_utf8_lossy(&x)
+                .split(' ')
+                .filter(|path| is_writable(Path::new(path)))
+                .map(|x| x.to_owned())
+                .collect::<Vec<_>>();
+            format!(
+                "{}/_forc",
+                paths.first().expect("No writable path found for zsh")
+            )
+        }
+        _ => format!("{}/.forc.autocomplete", dir),
+    };
+
+    let mut file = File::create(&forc_autocomplete_path).unwrap_or_else(|_| {
+        panic!("Cannot write to the autocomplete file in path {forc_autocomplete_path}")
+    });
+    generate_autocomplete_script(target, &mut file);
+
+    // Check if the shell config file already has the forc completions. Some shells do not require
+    // this step, therefore this maybe None
+    let user_shell_config = match target {
+        Target::Fish => Some(format!("{}/.config/fish/config.fish", dir)),
+        Target::Elvish => Some(format!("{}/.elvish/rc.elv", dir)),
+        Target::PowerShell => Some(format!(
+            "{}/.config/powershell/Microsoft.PowerShell_profile.ps1",
+            dir
+        )),
+        Target::Bash => Some(format!("{}/.bashrc", dir)),
+        Target::Fig => Some(format!("{}/.config/fig/fig.fish", dir)),
+        _ => None,
+    };
+
+    if let Some(file_path) = user_shell_config {
+        // Update the user_shell_config
+        let file = File::open(&file_path).expect("Open the shell config file");
+        let reader = BufReader::new(file);
+
+        for line in reader.lines().map_while(Result::ok) {
+            if line.contains(&forc_autocomplete_path) {
+                println!("Forc completions is already installed");
```

Comment:
```
                println!("Forc completions is already installed for {shell}");
```

Reply:
If the completions script has been updated, the user should not see this message. They should see something like "Successfully updated for completions."